packages/@aws-cdk/aws-codebuild/test/test.project.ts

```diff
@@ -1466,12 +1475,236 @@ export = {
+
+        // THEN
+        expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+          'PolicyDocument': {
+            'Statement': arrayWith({
+              'Action': 'secretsmanager:GetSecretValue',
+              'Effect': 'Allow',
+              'Resource': {
+                'Fn::Join': ['', [
+                  'arn:',
+                  { Ref: 'AWS::Partition' },
+                  ':secretsmanager:',
+                  { Ref: 'AWS::Region' },
+                  ':',
+                  { Ref: 'AWS::AccountId' },
+                  ':secret:secretId-??????',
+                ]],
+              },
+            }),
+          },
+        }));
+
+        test.done();
+      },
+
+      'can be provided as a SecretValue with a jsonField and versionId'(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+        // WHEN
+        const secretValue = cdk.SecretValue.secretsManager(
+          'arn:aws:secretsmanager:us-west-2:123456789012:secret:my-secret-123456',
+          { jsonField: 'json-key', versionId: 'version-id' },
+        );
+        new codebuild.PipelineProject(stack, 'Project', {
+          environmentVariables: {
+            'ENV_VAR1': {
+              type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
+              value: secretValue,
+            },
+          },
+        });
+
+        // THEN
+        expect(stack).to(haveResourceLike('AWS::CodeBuild::Project', {
+          'Environment': {
+            'EnvironmentVariables': [
+              {
+                'Name': 'ENV_VAR1',
+                'Type': 'SECRETS_MANAGER',
+                'Value': 'arn:aws:secretsmanager:us-west-2:123456789012:secret:my-secret-123456:json-key:version-id',
+              },
+            ],
+          },
+        }));
+
+        // THEN
+        expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+          'PolicyDocument': {
+            'Statement': arrayWith({
+              'Action': 'secretsmanager:GetSecretValue',
+              'Effect': 'Allow',
+              'Resource': 'arn:aws:secretsmanager:us-west-2:123456789012:secret:my-secret-123456*',
+            }),
+          },
+        }));
+
+        test.done();
+      },
+
+      'throws error when SecretValue is provided with versionStage and versionId'(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+
+        // WHEN
+        const secretValue = cdk.SecretValue.secretsManager('secretId', { versionStage: 'version-stage', versionId: 'version-id' });
+
+        // THEN
+        test.throws(() => {
+          new codebuild.PipelineProject(stack, 'Project', {
+            environmentVariables: {
+              'ENV_VAR1': {
+                type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
+                value: secretValue,
+              },
+            },
+          });
+        }, /Secret Value cannot provide versionStage and versionId at the same time!/);
+
+        test.done();
+      },
```

Comment:
We are missing an important test case here: providing a `secret.secretValue`, where `secret` is a new Secret, created in a Stack that is in a different account than the CodeBuild Project's Stack.

Reply:
Hm for this we would expect the Partition and Region to also come from the SecretStack right?

Reply:
You tell me 🙂. I would assume it has to, otherwise the KMS permissions will be incorrect I think.

Reply:
Hm seems like we hit another underlying problem. When creating a new Secret, the secretArn get parsed here:
https://github.com/aws/aws-cdk/blob/cdf02649bb589e447f84ed52b6f68ed5932b5edd/packages/%40aws-cdk/aws-secretsmanager/lib/secret.ts#L462-L467
We are not parsing the Partition nor the Region here. Should we fix this in this PR?

Reply:
Actually, passing the partition, account and region there will not do anything (feel free to try, and let me know if it fixes the problem - but my guess would that it doesn't).

The problem is that the ARN is actually a `{ Ref: Secret }` expression. The problem might be with the `secretValue` method in `Secret` - you might have to pass the partition, account and region there.

Reply:
Actually, I think we might need one more test - importing a Secret using `fromSecretNameV2()` in a Stack that is in a different account than the Stack the CodeBuild Project belongs to.

Reply:
Seems like we were overthinking this problem yesterday. `Secrets` should live in the same region from where you try to retrieve if and for multi region setup one would use replicas. So I think everything works as expected. Will add the tests now.

Reply:
Alright those test cases were really good and brought up some more code changes.
Not sure if `SecretQualifier` is still the best name when we now save the account here as well. What do you think?